Classes/RedirectStorage.php

```diff
@@ -165,15 +165,18 @@ protected function updateDependingRedirects(RedirectInterface $newRedirect)
         $existingRedirectForTargetUriPath = $this->redirectRepository->findOneBySourceUriPathAndHost($newRedirect->getTargetUriPath(), $newRedirect->getHost(), false);
 
         if ($existingRedirectForTargetUriPath !== null) {
-            if ($existingRedirectForTargetUriPath->getTargetUriPath() === $newRedirect->getSourceUriPath()) {
-                $this->redirectRepository->remove($existingRedirectForTargetUriPath);
-            } else {
-                throw new Exception(sprintf('A redirect exists for the target URI path "%s", please remove it first.', $newRedirect->getTargetUriPath()), 1382091526);
-            }
+            $this->persistenceManager->whitelistObject($existingRedirectForTargetUriPath);
+            $this->redirectRepository->remove($existingRedirectForTargetUriPath);
+            $this->persistenceManager->persistAll(true);
+            $this->_logger->log(sprintf('A redirect exists for the target URI path "%s", removed automatically.', $newRedirect->getTargetUriPath()), LOG_NOTICE);
         }
         if ($existingRedirectForSourceUriPath !== null) {
-            throw new Exception(sprintf('A redirect exists for the source URI path "%s", please remove it first.', $newRedirect->getSourceUriPath()), 1382091456);
+            $this->persistenceManager->whitelistObject($existingRedirectForSourceUriPath);
```

Comment:
could we take this code here and put it into a separate helper method? I know it's not 100% identical to line 168-171, but it could easily be